Reserves a memory to avoid a memory reallocation caused by push_back.
diff --git src/PyZyDatabase.cc src/PyZyDatabase.cc
index 064f54a..c9769b2 100644
--- src/PyZyDatabase.cc
+++ src/PyZyDatabase.cc
@@ -51,6 +51,9 @@ public:
 
     void double_ (void) {
         size_t i = size ();
+        // To avoid a invalid referece caused by a memory reallocation, which
+        // may occur on push_back, we call reserve here.
+        reserve (i * 2);
         do {
             push_back (at (--i));
         } while (i > 0);
@@ -58,8 +61,11 @@ public:
 
     void triple (void) {
         size_t i = size ();
+        // To avoid a invalid referece caused by a memory reallocation, which
+        // may occur on push_back, we call reserve here.
+        reserve (i * 3);
         do {
-            const std::string & value = std::vector<std::string>::at (--i);
+            const std::string & value = at (--i);
             push_back (value);
             push_back (value);
         } while (i > 0);
